test/e2e/common/configmap.go

```diff
@@ -212,10 +236,17 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
 		}
 		framework.ExpectEqual(testConfigMapFound, true, "failed to find ConfigMap in list")
 
+		ginkgo.By("deleting the ConfigMap by a collection")
 		err = f.ClientSet.CoreV1().ConfigMaps(testNamespaceName).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{
 			LabelSelector: "test-configmap-static=true",
 		})
 		framework.ExpectNoError(err, "failed to delete ConfigMap collection with LabelSelector")
+		ginkgo.By("waiting for the ConfigMap to be deleted")
+		for watchEvent := range resourceWatchChan {
+			if watchEvent.Type == watch.Deleted {
+				break
+			}
+		}
```

Comment:
I think this applies for all of the watch-based checks, but it's sticking out to me here.  Let's say the ConfigMap isn't deleted within 60 seconds, how will this test catch the failure?

In other words, what does the failure of this test look like if the watch times out?

Reply:
I'm not sure that anything will happen on watch timeout. I tried with a timeout of both `0` and `1` in different local tests and this test still succeeded.

Anyhow, I'm not sure how to implement handling of failure (but would also like that to be in there) - are there any places in the test folder where you'd possibly please be able to provide me with an example of how to handle failure on resource deletion?

Reply:
I also asked in #k8s-conformance slack https://kubernetes.slack.com/archives/C78F00H99/p1585767344008100

Reply:
At a minimum this should have a flag set to false before the loop, update the flag to true on the inside of the loop, and do a framework.Expect(...) after the loop, eg:
```
foundWatchDeleteEvent := false
for ... {
 if ... {
  foundWatchDeleteEvent := true
  break
  }
}
framework.ExpectEqual(true, foundWatchDeleteEvent, "expected to find a watch.Delete event configmap %s", configMap)
```

Reply:
Added in 7ba8b35b9c4